packages/node_modules/@ciscospark/widget-recents/src/container.js

```diff
@@ -638,14 +676,24 @@ export class RecentsWidget extends Component {
       });
 
       let filteredSpacesListWithActivityText;
+      // Filter space list based on type
+      const groupBy = spaceTypeFilter === SPACE_TYPE_GROUP || spaceTypeFilter === SPACE_TYPE_ONE_ON_ONE;
```

Comment:
Doing this all in the render function is going to take a TON of compute power for every render. I think it might be best to move this into the `getRecentSpaces` function in `selector.js`. That way our results are memoized and we don't have to compute them every time the widget renders.